src/Template/Bake/Model/table.ctp

```diff
@@ -13,6 +13,29 @@
  * @license       http://www.opensource.org/licenses/mit-license.php MIT License
  */
 use Cake\Utility\Inflector;
+
+$annotations = [];
+foreach ($associations as $type => $assocs)
+{
+    foreach ($assocs as $assoc)
+    {
+        $typeStr = Inflector::camelize($type);
+        $annotations[] = "@property \Cake\ORM\Association\\{$typeStr} \${$assoc['alias']}";
+    }
+}
+if(PHP_MAJOR_VERSION >= 7)
+{
+    $annotations[] = "@method get(\$primaryKey, \$options = []) : {$entity}";
+    $annotations[] = "@method newEntity(\$data = null, array \$options = []) : {$entity}";
+    $annotations[] = "@method newEntities(array \$data, array \$options = []) : {$entity}[]";
+    $annotations[] = "@method save(EntityInterface \$entity, \$options = []) : {$entity}";
+    $annotations[] = "@method patchEntity(EntityInterface \$entity, array \$data, array \$options = []) : {$entity}";
+    $annotations[] = "@method patchEntities(\$entities, array \$data, array \$options = []) : {$entity}[]";
+    foreach ($behaviors as $behavior => $behaviorData)
+    {
```

Comment:
@josegonzalez On PHP7 its written like this `@method foo() : array` and it doesn't work in PhpStorm is the version is set to PHP 5. But I think if I write like this `@method array foo()` it works in all versions. I'll fix it tomorrow.